core/network/source_other.go

```diff
@@ -38,9 +112,13 @@ func NewNetAddr(a string) *netAddr {
 	if ip == nil {
 		ip = net.ParseIP(a)
 	}
-	res.ip = ip
 
-	return res
+	if ip == nil {
```

Comment:
This checking for validity was requested by @achilleasa in a pre-cursor to this patch.